src/Compilers/CSharp/Test/Symbol/Symbols/RequiredMembersTests.cs

```diff
@@ -0,0 +1,1181 @@
+        );
+    }
+
+    [Fact]
+    public void LangVersion()
+    {
+        var comp = CreateCompilationWithRequiredMembers(@"
+class C
+{
+    internal required int Field;
+    internal required int Prop { get; set; }
+}
+", parseOptions: TestOptions.Regular10);
+
+        comp.VerifyDiagnostics(
+            // (4,27): error CS8652: The feature 'required members' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+            //     internal required int Field;
+            Diagnostic(ErrorCode.ERR_FeatureInPreview, "Field").WithArguments("required members").WithLocation(4, 27),
+            // (4,27): warning CS0649: Field 'C.Field' is never assigned to, and will always have its default value 0
+            //     internal required int Field;
+            Diagnostic(ErrorCode.WRN_UnassignedInternalField, "Field").WithArguments("C.Field", "0").WithLocation(4, 27),
+            // (5,27): error CS8652: The feature 'required members' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+            //     internal required int Prop { get; set; }
+            Diagnostic(ErrorCode.ERR_FeatureInPreview, "Prop").WithArguments("required members").WithLocation(5, 27)
+        );
+    }
+
+    [Fact]
+    public void DuplicateKeyword()
+    {
+        var comp = CreateCompilationWithRequiredMembers(@"
+class C
+{
+    internal required required int Field;
+    internal required required int Prop { get; set; }
+}
+");
+
+        comp.VerifyDiagnostics(
+            // (4,23): error CS1004: Duplicate 'required' modifier
+            //     internal required required int Field;
+            Diagnostic(ErrorCode.ERR_DuplicateModifier, "required").WithArguments("required").WithLocation(4, 23),
+            // (4,36): warning CS0649: Field 'C.Field' is never assigned to, and will always have its default value 0
+            //     internal required required int Field;
+            Diagnostic(ErrorCode.WRN_UnassignedInternalField, "Field").WithArguments("C.Field", "0").WithLocation(4, 36),
+            // (5,23): error CS1004: Duplicate 'required' modifier
+            //     internal required required int Prop { get; set; }
+            Diagnostic(ErrorCode.ERR_DuplicateModifier, "required").WithArguments("required").WithLocation(5, 23)
+        );
+    }
+
+    [Theory]
+    [CombinatorialData]
+    public void InvalidNames(bool use10)
+    {
+        var comp = CreateCompilationWithRequiredMembers(@"
+namespace N1
+{
+    struct required {}
+}
+namespace N2
+{
+    class required {}
+}
+namespace N3
+{
+    interface required {}
+}
+namespace N4
+{
+    delegate void required();
+}
+namespace N5
+{
+    record required();
+}
+namespace N6
+{
+    record struct required();
+}
+namespace N7
+{
+    class C
+    {
+        class required {}
+    }
+}
+namespace N8
+{
+    class required<T> {}
+}
+", parseOptions: use10 ? TestOptions.Regular10 : null);
```

Comment:
This test should use `RegularNext` explicitly for the alternative, rather than preview.